rwengine/src/engine/GameData.cpp

```diff
@@ -62,102 +56,81 @@ void GameData::load()
 	loadTXD("hud.txd");
 	loadTXD("fonts.txd");
 	
-	loadCarcols(datpath+"/data/carcols.dat");
-	loadWeather(datpath+"/data/timecyc.dat");
-	loadHandling(datpath+"/data/handling.cfg");
-	loadWaterpro(datpath+"/data/waterpro.dat");
-	loadWater(datpath+"/data/water.dat");
-	loadWeaponDAT(datpath+"/data/weapon.dat");
+	loadCarcols("data/carcols.dat");
+	loadWeather("data/timecyc.dat");
+	loadHandling("data/handling.cfg");
+	loadWaterpro("data/waterpro.dat");
+	loadWeaponDAT("data/weapon.dat");
 
 	loadIFP("ped.ifp");
 }
 
-void GameData::parseDAT(const std::string& path)
+void GameData::loadLevelFile(const std::string& path)
 {
-	std::ifstream datfile(path.c_str());
-	
+	auto datpath = index.findFilePath(path);
+	std::ifstream datfile(datpath.c_str());
+
 	if(!datfile.is_open()) 
 	{
 		logger->error("Data", "Failed to open game file " + path);
+		return;
 	}
-	else
+
+	for(std::string line, cmd; std::getline(datfile, line);)
 	{
-		for(std::string line, cmd; std::getline(datfile, line);)
+		if(line.size() == 0 || line[0] == '#') continue;
+		#ifndef RW_WINDOWS
+			line.erase(line.size()-1);
+		#endif
+
+		size_t space = line.find_first_of(' ');
+		if(space != line.npos)
 		{
-			if(line.size() == 0 || line[0] == '#') continue;
-			#ifndef RW_WINDOWS
-				line.erase(line.size()-1);
-			#endif
-			
-			size_t space = line.find_first_of(' ');
-			if(space != line.npos)
+			cmd = line.substr(0, space);
+			if(cmd == "IDE")
 			{
-				cmd = line.substr(0, space);
-				if(cmd == "IDE")
-				{
-					addIDE(line.substr(space+1));
-				}
-				else if(cmd == "SPLASH")
-				{
-					splash = line.substr(space+1);
-				}
-				else if(cmd == "COLFILE")
-				{
-					int zone  = atoi(line.substr(space+1,1).c_str());
-					std::string file = line.substr(space+3);
-					loadCOL(zone, file);
-				}
-				else if(cmd == "IPL")
-				{
-					std::string fixedpath = fixPath(line.substr(space+1));
-					fixedpath = findPathRealCase(datpath, fixedpath);
-					loadIPL(fixedpath);
-				}
-				else if(cmd == "TEXDICTION") 
-				{
-					std::string texpath = line.substr(space+1);
-					for( size_t t = 0; t < texpath.size(); ++t) {
-						texpath[t] = tolower(texpath[t]);
-						if(texpath[t] == '\\') {
-							texpath[t] = '/';
-						}
-					}
-					std::string texname = texpath.substr(texpath.find_last_of("/")+1);
-					loadTXD(texname);
-				}
+				auto path = line.substr(space+1);
+				auto systempath = index.findFilePath(path).native();
+				loadIDE(systempath);
```

Comment:
This resolve-before-call style is at different from the way the other loadX functions are implemented. Should be consistent.